src/main/java/org/example/StudentDAO.java

```diff
@@ -0,0 +1,51 @@
+package org.example;
+
+import org.example.entities.Student;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import java.util.List;
+
+public class StudentDAO {
+    public void addStudent(Student student) {
+        SessionFactory sessionFactory = HibernateSession.getSessionFactory();

```

Comment:
Would be better to create SessionFactory on the class level and set it through a constructor. Your approach duplicates the same code across all methods in this class.